Tune MariaDB settings for better performance

khurtado opened the pull request:
Tune MariaDB settings for better performance
More details: https://github.com/dmwm/WMCore/issues/11977

The performance with these settings increased from -5 to 30% with these changes (comparing docker deployment on Alma9 vs CC7 RPM host-based deployment):
https://gitlab.cern.ch/dmwm/wmcore-docs/-/blob/master/docs/databases/MariaDB-benchmark.md

amaltaro commented:
@khurtado can this 8GB pool be a problem when we run MariaDB in small nodes (e.g. with <= 8GB of memory RAM)?

I would suggest linking the tests documentation to this PR as well.

todor-ivanov reviewed (approved):
looks good

khurtado commented:
@amaltaro That's a good point. I added some commented lines and updated the documentation for it.

An alternative to having the users uncomment settings for small nodes would be to create another branch specially for testbed hardware, but if this is the only change needed in general, for the containerized deployment, then I don't think it is necessary right now.

amaltaro commented on docker/pypi/wmagent-mariadb/README.md:
````diff
@@ -67,6 +67,18 @@ local/mariadb              latest    4efa646aea3e   6 minutes ago    950MB
 ./mariadb-docker-build.sh -t 10.6.5 -p
 ```
 
+#### Building MariaDB on small nodes (e.g.: testbed)
````
Perhaps change from "Building" to "Running"?

amaltaro commented on docker/pypi/wmagent-mariadb/my.cnf:
```diff
@@ -46,9 +46,19 @@ innodb_checksum_algorithm=full_crc32
+
+# Production nodes
+# Settings based on benchmark tests:
+# https://gitlab.cern.ch/dmwm/wmcore-docs/-/blob/master/docs/wmcore/MariaDB-benchmark-tests.md
```
This link needs to be fixed.

amaltaro reviewed (changes requested):
@khurtado Kenyi, in addition to the few comments along the code, you might want to add to the PR description that an increase of X% is seen with such configuration (keeping a link to the full tests and results, as you already did).

khurtado commented:
> @khurtado Kenyi, in addition to the few comments along the code, you might want to add to the PR description that an increase of X% is seen with such configuration (keeping a link to the full tests and results, as you already did).

Done!

amaltaro commented:
Thanks Kenyi!
